### DISCUSSION TITLE
Starting projects from scratch

### DISCUSSION THREAD

**Discussion by oliverjam** [foundersandcoders/coursebook#582] [Ideas]
Are we limiting potential learning by making teams start their projects totally from scratch every week? We might get better results having devs focus on implementing features within an existing structure.

This is both more realistic (not sure I've ever started a project from scratch at a professional job), and would allow them to focus their time on all aspect of a single feature (implementation, styling, tests etc) rather than being spread thin doing server/DB boilerplate setup for the fourth time.

In my opinion the biggest benefit would be more interesting projects. Right now I feel like people are a little demotivated by building the same uninteresting set of features for several weeks in a row. If we provide a starting point they'll be able to go much further with their ideas.

**Comment by sofer** [comment]
It's worth remembering that we have done this before across two weeks. it became an issue with technical debt piling up very quickly in the second week and, as I recall, teams wrestling with breaking code from the previous week and unable to implement features on top of incomplete user stories. 

That doesn't mean it's a bad idea, it just means that it has to be done with care. Teams probably need to stay together for the length of the multi-week project, the core user stories need to be achievable (and achieved) and the stretch goals not needed for subsequent weeks.

There is an interesting question, too, about how we might structure this. Weeks 1-3 could perhaps be restructured as a single project. Week 4 could probably be coupled with either Week 5 or 6, but possibly not both unless you want people to re-implement the same features in different frameworks.

**Reply by oliverjam** [reply]
Sorry if I wasn't clear: I'm not suggesting we let them work on the same project across weeks. I'm suggesting we create functioning projects then ask the teams to implement features _within_ those projects.

**Comment by Albadylic** [comment]
It's an interesting question. I feel like the main benefit to starting from scratch is that fundamental understanding of how all the pieces fit together and how to create the foundations.

However, as we move higher up the ladder of abstraction, the fundamental underpinnings become less relevant. I agree with you that testing and styling seem to be the often forgotten aspects of a project and that being able to actually spend time on these would be valuable.

I also feel like you're not suggesting this for every project, but moreso for the later ones once they've got a reasonable underpinning.

It might be worth thinking about the final projects too. A lot of the time is spent setting up technologies and getting everything set-up in the first week. Having a component library or a set of starter projects to build upon could be beneficial on building more robust and complex projects.

I'd be a little concerned that the time spent setting up would be instead spent getting a grasp on how the codebase works. Or that how the whole thing really works is less well understood.
